src/nccl_ofi_net.c

```diff
@@ -1197,6 +1197,17 @@ static ncclResult_t update_nccl_topology()
 	return ret;
 }
 
+static void ofi_read_num_gpus(int *count){
+	char *fi_efa_num_gpus = getenv("FI_EFA_NUM_GPUS");
```

Comment:
`FI_EFA` prefix is used for EFA's libfabric library. So, we shouldn't be naming this variables with this prefix. Why do we need `FI_EFA_NUM_GPUS` rather than relying on `cudaGetDeviceCount`?

Reply:
This is to handle the scenarios to all the GPU's are not visible to the process. This happens when `CUDA_VISIBLE_DEVICES ` environment variable is set when launching the process. PyTorch XLA leverages this to make only one gpu visible to each of the training process in case of distributed training.

Reply:
But my understanding is that when `CUDA_VISIBLE_DEVICES` is set, then `cudaGetDeviceCount` would return the number of visible devices. Isn't that true for your runs?